### ISSUE TITLE
Replace file entries with gitAUTOINC and svnX are incompatible from 1.9 to 1.10 

### ISSUE THREAD

**Issue by bmos1** [matthewb66/import_yocto_bm#2]
Hi Matthew,

first of all best greetings from Commend. We appreciate your work. The recent update from version 1.9 to 1.10 brought up an compatibility issue when the replace file contains gitAUTOINC and svnX entries. It seems that 1.10 modifies the mapping string by prepending "version+gitX-" respectively "version+svnX-".  Is this intended new behavior or a typo in the script?

Attached please see the diff example:

`
/oss/blackduck$ diff replace_file_v1.9 replace_file_v1.10
RECIPE meta-oe/protobuf/3.5.1+gitAUTOINC+106ffc04be-r0 meta-oe/protobuf/3.5.1+gitX-r0
RECIPE meta-oe/protobuf/**3.5.1+gitX-** 3.5.1+gitAUTOINC+106ffc04be-r0 meta-oe/protobuf/3.5.1+gitX-r0

RECIPE meta-cos/avrdude/6.3+svn1425-r2 meta-maker/avrdude/6.3+svnX-r0
RECIPE meta-cos/avrdude/**6.3+svnX-** 6.3+svn1425-r2 meta-maker/avrdude/6.3+svnX-r0
`

**Comment by bmos1** [comment]
This is a patch file to fix the issue

[0001-Bugfix-Repair-gitAUTOINC-and-svnX-to-be-compatible-w.patch.txt](https://github.com/matthewb66/import_yocto_bm/files/6143250/0001-Bugfix-Repair-gitAUTOINC-and-svnX-to-be-compatible-w.patch.txt)

**Comment by matthewb66** [comment]
Yes this was an intended change as the previous script would not allow replace to work on recipes which contained GitX or SvnX. This is because it replaced strings before they were compared with the replace file making translation impossible.

I can't understand from your description what is the issue with the improved script for you 

Please send me an email at my work address with all information including;
License manifest file, output of bitbake-layers show-recipes and a full description of what you are trying to do and why.

**Comment by bmos1** [comment]
Thanks for your fast reply. I tried the new AUTOINC feature, but found an implementation bug.
This is why I provided to you a patch file via GitHub. A fork and a pull request would be to much for small changes on a single file.

Why do provide a patch?
My replace file from v1.9 was not compatible and didn’t replace anything with AUTOINC or +svn the name.

Replace file:
RECIPE meta-oe/protobuf/3.5.1+gitAUTOINC+106ffc04be-r0 meta-oe/protobuf/3.5.1+gitX-r0
Console output:
- Component meta-oe/protobuf/3.5.1+gitX-3.5.1+gitAUTOINC+106ffc04be-r0: Recipe exists in KB within the layer but version does not 

How I made the change?
The code changes for v1.10 expect a revision suffix e.g. “-r0”. The problem is that it comes from “recipeinfo” file afterwards.
Hence recipes[recipe].split("-")[-1] doesn’t return the revision only but instead return “3.5.1+gitAUTOINC+106ffc04be-r0”
Finally recipes[recipe] = 3.5.1+gitX-3.5.1+gitAUTOINC+106ffc04be-r0 is wrong.

What did I change?
Hence I moved the following code after revision has been set by “recipeinfo”. 
In case of args.debug the suffix is set to “-r0”.

if recipes[recipe].find("AUTOINC") != -1:
    recipes[recipe] = recipes[recipe].split("AUTOINC")[0] + "X-" + recipes[recipe].split("-")[-1]
if recipes[recipe].find("+svn") != -1:
    recipes[recipe] = recipes[recipe].split("+svn")[0] + "+svnX-" + recipes[recipe].split("-")[-1]
